src/tests/ftest/util/network_utils.py

```diff
@@ -0,0 +1,559 @@
+#!/usr/bin/python3
+"""
+  (C) Copyright 2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import errno
+from logging import getLogger
+import os
+import re
+
+from ClusterShell.NodeSet import NodeSet
+
+from exception_utils import CommandFailure
+from general_utils import run_task, display_task
+
+SUPPORTED_PROVIDERS = (
+    "ofi+sockets", "ofi+tcp", "ofi+tcp;ofi_rxm", "ofi+verbs", "ofi+verbs;ofi_rxm", "ucx+dc_x")
+
+
+class NetworkDevice():
+    """A class to represent the information of a network device."""
+
+    def __init__(self, host, device, ib_device, port, provider, numa):
+        """Initialize the network device data object."""
+        self.host = host
+        self.device = device
+        self.ib_device = ib_device
+        self.port = port
+        self.provider = provider
+        self.numa = numa
+
+    def __repr__(self):
+        """Overwrite to display formatted devices."""
+        return self.__str__()
+
+    def __str__(self):
+        """Overwrite to display formatted devices."""
+        settings = ["{}={}".format(key, getattr(self, key, None)) for key in self.__dict__]
+        return "NetworkDevice({})".format(", ".join(settings))
+
+    def __ne__(self, other):
+        """Override the default not-equal implementation."""
+        return not self.__eq__(other)
+
+    def __eq__(self, other):
+        """Override the default implementation to compare devices."""
+        status = isinstance(other, NetworkDevice)
+        for key in self.__dict__:
+            if not status:
+                break
+            try:
+                status &= getattr(self, key) == getattr(other, key)
+            except AttributeError:
+                status = False
+        return status
```

Comment:
Optional - I realize you only moved this function, but this implementation bothers me. I think just a simple `==` should work here, and is more complete
```suggestion
    def __eq__(self, other):
        """Override the default implementation to compare devices."""
        if not isinstance(other, NetworkDevice):
            return False
        return self.__dict__ == other.__dict__
```

Reply:
I'm missing an additional edit here, so I do need a bit of extra code, but I can clean it up.